gravity.sh

```diff
@@ -591,8 +592,40 @@ gravity_DownloadBlocklistFromUrl() {
     fi
   fi
 
-  # shellcheck disable=SC2086
-  httpCode=$(curl --connect-timeout ${curl_connect_timeout} -s -L ${compression} ${cmd_ext} ${heisenbergCompensator} -w "%{http_code}" "${url}" -o "${listCurlBuffer}" 2>/dev/null)
+  # If we are going to "download" a local file, we first check if the target
+  # file has a+r permission. We explicitly check for all+read because we want
+  # to make sure that the file is readable by everyone and not just the user
+  # running the script.
+  if [[ $url == "file://"* ]]; then
+    # Get the file path
+    file_path=$(echo "$url" | cut -d'/' -f3-)
+    # Check if the file exists and is a regular file (or a symlink to one)
+    if [[ ! -e $file_path ]]; then
+      # Output that the file does not exist
+      echo -e "${OVER}  ${CROSS} ${file_path} does not exist"
+      download=false
+    elif [[ ! -f $file_path ]]; then
+      # Output that the file is not a regular file
+      echo -e "${OVER}  ${CROSS} ${file_path} is not a regular file"
+      download=false
+    else
+      # Check if the file has a+r permissions
+      permissions=$(stat -Lc "%a" "$file_path")
+      if [[ $permissions == *4 || $permissions == *5 || $permissions == *6 || $permissions == *7 ]]; then
+        # Output that we are using the local file
+        echo -e "${OVER}  ${INFO} Using local file ${file_path}"
+      else
+        # Output that the file does not have the correct permissions
+        echo -e "${OVER}  ${CROSS} Cannot read file (file needs to have o+r permission)"
```

Comment:
```suggestion
        echo -e "${OVER}  ${CROSS} Cannot read ${file_path} (file needs to have o+r permission, file is set to ${permissions}) "
```

Reply:
I didn't do this as it seemed odd to have `o+r` in the front but then octal in the back. Also there is no need to re-print the filename here as it is already printed right above and - IMO - just makes the warning line very long without adding any further useful content

Reply:
Okay, I see the octal vs regular situation but I don't see where the filename is printed when the file is not being used. I see it on line 616 but that is for when the file is being used.